setup.py

```diff
@@ -1,48 +1,86 @@
 from __future__ import absolute_import, division, print_function
 
-from setuptools import setup, find_packages, Extension
+from distutils.command.sdist import sdist as _sdist
+import glob
 import os
+import sys
+from setuptools import setup, find_packages, Extension
+from setuptools.command.build_ext import build_ext as _build_ext
 
-import numpy as np
-
-from Cython.Build import cythonize
+PACKAGE_NAME = 'stratify'
+PACKAGE_DIR = os.path.abspath(os.path.dirname(__file__))
 
 
-NAME = 'stratify'
-DIR = os.path.abspath(os.path.dirname(__file__))
+try:
+    # Detect if Cython is available. Where it is, use it, otherwise fall back
+    # to C source included with source distribution.
+    from Cython.Build import cythonize
+except ImportError:
+    cythonize = None
 
-extension_kwargs = {'include_dirs': [np.get_include()]}
+source_suffix = '.pyx' if cythonize else '.c'
+extension_kwargs = {}
+cython_directives = {'binding': True}
 cython_coverage_enabled = os.environ.get('CYTHON_COVERAGE', None)
-if cython_coverage_enabled:
+if cythonize and cython_coverage_enabled:
     extension_kwargs.update({'define_macros': [('CYTHON_TRACE_NOGIL', '1')]})
+    cython_directives.update({'linetrace': True})
+
+extensions = []
+for source_file in glob.glob('{}/*{}'.format(PACKAGE_NAME, source_suffix)):
+    source_file_nosuf, _ = os.path.splitext(os.path.basename(source_file))
+    extensions.append(
+        Extension('{}.{}'.format(PACKAGE_NAME, source_file_nosuf),
+                  sources=['{}/{}{}'.format(
+                      PACKAGE_NAME, source_file_nosuf, source_suffix)],
+                  **extension_kwargs))
+
+if cythonize and 'clean' not in sys.argv:
+    extensions = cythonize(extensions, compiler_directives=cython_directives)
+
+
+class SDist(_sdist):
+    # Source distribution build runs Cython so that Cython is not needed as
+    # an install dependency.
+    def run(self):
+        cythonize(extensions, compiler_directives=cython_directives)
```

Comment:
@tjtg In the case where `python setup.py sdist` is run with no `cython` installed, we'd want:
```python
if cythonize:
    cythonize(extensions, compiler_directives=cython_directives)
```
Otherwise, you'll get a `TypeError: 'NoneType' object is not callable`